### ISSUE TITLE
Rails 3.2 support

### ISSUE THREAD

**Issue by francesc** [francesc/rails-translate-routes#7]
Just checked out Rails 3.2 it seems the routing recognition is replaced by Journey engine (https://github.com/rails/journey/tree/master/lib/journey) and have many differences with the current one, so probably I'll have to make two branches rails 3.0-3.1 as 0.x releases and a new one for >3.2 as 1.x release.

Haven't looked deeply on it but it seems it will take a while as a major rewrite will be needed :(

**Comment by ncri** [comment]
I just patched 2 functions to get rails 3.2 support in my app:

```
def translate route_set
  Rails.logger.info "Translating routes (default locale: #{default_locale})" if defined?(Rails) && defined?(Rails.logger)

  # save original routes and clear route set
  original_routes = route_set.routes.dup                     # Array [routeA, routeB, ...]
  original_routes.routes.delete_if{|r| r.path.spec.to_s == '/assets'  }
  original_named_routes = route_set.named_routes.routes.dup  # Hash {:name => :route}

  translated_routes = []
  original_routes.each do |original_route|
    translations_for(original_route).each do |translated_route_args|
      translated_routes << translated_route_args
    end
  end

  reset_route_set route_set

  translated_routes.each do |translated_route_args|
    route_set.add_route *translated_route_args
  end

  original_named_routes.each_key do |route_name|
    route_set.named_routes.helpers.concat add_untranslated_helpers_to_controllers_and_views(route_name)
  end

  if root_route = original_named_routes[:root]
    add_root_route root_route, route_set
  end

end


# Generate translation for a single route for one locale
def translate_route route, locale
  conditions = { :path_info => translate_path(route.path.spec.to_s, locale) }
  conditions[:request_method] = parse_request_methods route.verb if route.verb != //
  requirements = route.requirements.merge LOCALE_PARAM_KEY => locale
  defaults = route.defaults.merge LOCALE_PARAM_KEY => locale.dup
  new_name = "#{route.name}_#{locale_suffix(locale)}" if route.name

  [route.app, conditions, requirements, defaults, new_name]
end
```

Hope this helps!  Disclaimer: not thoroughly tested yet.

**Comment by francesc** [comment]
Hi Nico,

Just make a quick test and It works almost everything except adding the root path when using: prefix_on_default_locale => true. I'll try to figure it out tomorrow, and make a version compatible for <3.2 && >3.2 in a single gem. Hopefully if we don't find any other bug we can make a new release this week :)

Thanks for your contribution!

**Comment by francesc** [comment]
Ok, already figured out the root thing, couldn't go to lunch before trying to solve it :D

If somebody wants to try it replace add_root_route with:

```
def add_root_route route, route_set
  if @prefix_on_default_locale
    conditions = { :path_info => route.path.spec.to_s }
    conditions[:request_method] = parse_request_methods route.verb if route.verb != //
    route_set.add_route route.app, conditions, route.requirements, route.defaults, route.name
  end
end
```

**Comment by ncri** [comment]
Gr8! ;-)

**Comment by msamoylov** [comment]
Please please pleasu push you fix live :-) Itchy to use this app with 3.2

**Comment by ncri** [comment]
Just stick those three changed methods in an initializer if you cant wait for the release... ;-)

**Comment by francesc** [comment]
I'll push an "experimental" branch today so you'll be able to use it on your Gemfile ;)

**Comment by francesc** [comment]
Ok, just uploaded a quick branch for 3.2 ;) Just drop this line in your Gemfile to use it:

gem 'rails-translate-routes', :git => 'https://github.com/francesc/rails-translate-routes', :branch => 'rails32'

Note it's just a quick release, I'll have to make some modifications to the gem in order to make it compatible with 3.0-3.1 too.

**Comment by msamoylov** [comment]
Thank you!

**Comment by dedico** [comment]
Just in time - I got the error while updating my app to 3.2.1 :)

Thanks!

**Comment by francesc** [comment]
Released new gem version with rails 3.2 support ;)
